corehq/apps/userreports/reports/util.py

```diff
@@ -36,19 +39,38 @@ class ReportExport(object):
     """Export all the rows of a UCR report
     """
 
-    def __init__(self, domain, title, report_config, lang, filter_values):
+    def __init__(self, domain, title, report_config, lang, filter_values, request_user=None):
         self.domain = domain
         self.title = title
         self.report_config = report_config
         self.lang = lang
         self.filter_values = filter_values
+        self.request_user = request_user
 
     @property
-    @memoized
     def data_source(self):
         from corehq.apps.userreports.reports.data_source import ConfigurableReportDataSource
         data_source = ConfigurableReportDataSource.from_spec(self.report_config, include_prefilters=True)
         data_source.lang = self.lang
+        # Removing location from the filters for the locations that are not applicable for the current user.
+        if (toggles.LOCATION_RESTRICTED_SCHEDULED_REPORTS.enabled(self.domain)
+                and not self.request_user.has_permission(self.domain, 'access_all_locations')):
+            location_key = None
+            user_location_ids = self.request_user.get_location_ids(self.domain)
+            user_filtered_locations = []
+            for k, v in self.filter_values.items():
+                if 'computed_owner_location' in k:
```

Comment:
Where does this string `computed_owner_location` come from?

Reply:
When a report has filter `Case Owner (Location)` then those filter values in the report config comes as `computed_owner_location_xxx`
![image](https://github.com/dimagi/commcare-hq/assets/14318173/5ca372bb-f365-4d34-942b-5473d40d9244)

Reply:
Thanks for the detailed context!